lib/service-host.js

```diff
@@ -37,31 +37,28 @@ function getTypeScriptLanguageServiceHost(path, content) {
 }
 
 /**
- * Get a Language Service Host for Vue support, that has some extra methods needed by `@volar/vue-typescript`.
+ * Get a Language Service Host for Vue support.
  *
- * @typedef {import('@volar/vue-language-core/out/types').LanguageServiceHost} VueLanguageServiceHost
+ * @param {ts.CompilerOptions} compilerOptions compiler options
+ * @param {string} filepath file's name
+ * @param {string} code file's content
  *
- * @param {string} path path to file
- * @param {string} content file's content
- *
- * @returns {VueLanguageServiceHost}
+ * @returns {import('@vue/language-core').TypeScriptProjectHost}
  */
-function getVueLanguageServiceHost(path, content) {
-	const tsconfig = findTsconfig(path);
-	const vueCompilerOptions = getVueCompilerOptions(tsconfig);
-
+function getVueTypeScriptProjectHost(compilerOptions, filepath, code) {
 	return {
-		...getTypeScriptLanguageServiceHost(path, content),
-		getVueCompilationSettings: () => vueCompilerOptions,
-		/**
-		 * Can return either `require('typescript')` or `require('typescript/lib/tsserverlibrary')`.
-		 *
-		 * @see https://github.com/simonhaenisch/prettier-plugin-organize-imports/pull/60#discussion_r934408179
-		 *
-		 * @returns {any}
-		 */
-		getTypeScriptModule: () => ts,
+		...ts.sys,
+		configFileName: undefined,
+		getCurrentDirectory: () => '/',
+		getCompilationSettings: () => compilerOptions,
+		getScriptFileNames: () => [filepath],
+		getScriptSnapshot: (filename) => {
+			if (filepath === filename) return ts.ScriptSnapshot.fromString(code);
+		},
```

Comment:
These all seem to be part of the return value of `getTypeScriptLanguageServiceHost` already, why can't we pick them from there, like before?